lib/search/ui/widgets/search_bar.dart

```diff
@@ -0,0 +1,190 @@
+                    hintStyle: TextStyle(color: Theme.of(context).hintColor),
+                    border: InputBorder.none,
+                  ),
+                  onChanged: _onSearchChanged,
+                ),
+              ),
+              if (_searchController.text.isNotEmpty)
+                IconButton(
+                  icon: const Icon(Icons.cancel,
+                      color: ThemeColors.orangeDroidconColor),
+                  onPressed: _clearSearch,
+                ),
+            ],
+          ),
+        ),
+        const SizedBox(height: 10),
+
+        // Search Results
+        BlocBuilder<SearchCubit, SearchState>(
+          builder: (context, state) {
+            return state.when(
+              initial: () => Container(),
+              loading: () => const Center(child: CircularProgressIndicator()),
+              loaded: (results) => _buildSearchResults(results),
+              error: (message) => Center(
+                child: Text(
+                  message,
+                  style:
+                      const TextStyle(color: ThemeColors.orangeDroidconColor),
+                ),
+              ),
+            );
+          },
+        ),
+      ],
+    );
+  }
+
+  Widget _buildSearchResults(List<SearchResult> results) {
+    if (results.isEmpty) {
+      return const Text('No results found');
+    }
+
+    return ListView.builder(
+      shrinkWrap: true,
+      itemCount: results.length,
+      itemBuilder: (context, index) {
+        final result = results[index];
+        return Padding(
+          padding: const EdgeInsets.symmetric(vertical: 8.0),
+          child: Card(
+            shape: RoundedRectangleBorder(
+              borderRadius: BorderRadius.circular(20),
+              side: const BorderSide(color: ThemeColors.orangeDroidconColor),
+            ),
+            elevation: 4,
+            child: ListTile(
+              leading: SizedBox(
+                width: 48,
+                height: 48,
+                child: ClipRRect(
+                  borderRadius: BorderRadius.circular(12),
+                  child: ResolvedImage(imageUrl: result.imageUrl),
+                ),
+              ),
+              title: Text(
+                result.title,
+                style: const TextStyle(
+                  color: ThemeColors.blueDroidconColor,
+                  fontWeight: FontWeight.bold,
+                  fontSize: 14,
+                ),
+              ),
+              subtitle: Text(
+                result.subtitle,
+                style: const TextStyle(color: ThemeColors.greyTextColor),
+              ),
+              onTap: () => _handleResultTap(result),
+            ),
+          ),
+        );
+      },
+    );
+  }
+
+  void _handleResultTap(SearchResult result) {
+    FocusScope.of(context).unfocus();
+    switch (result.type) {
+      case SearchResultType.session:
+        if (result.session != null) {
+          context
```

Comment:
For uniformity purposes, please use
```dart
GoRouter.of(context).push
```
Ref: `lib/features/about/ui/organising_team.dart:41`

Reply:
done